panos_upgrade_assurance/check_firewall.py

```diff
@@ -1372,3 +1399,89 @@ def check_device_root_certificate_issue(self, fail_when_affected_version_only: b
             "expire December 31st, 2023."
         )
         return result
+
+    def check_cdss_and_panorama_certificate_issue(self):
+        """Checks whether the device is affected by the following advisory;
+
+        https://live.paloaltonetworks.com/t5/customer-advisories/additional-pan-os-certificate-expirations-and-new-comprehensive/ta-p/572158
+
+        Check will fail in either of following scenarios:
+
+         * Device is running an affected software version
+         * Device is running an affected content version
+         * Device is running the fixed content version or higher but has not been rebooted - note this is best effort,
+            and is based on when the content version was released and the device was rebooted
+
```

Comment:
Need returns docstring:
```suggestion

        # Returns

        CheckResult: Object of [`CheckResult`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkresult) class taking \
            value of:

        * [`CheckStatus.SUCCESS`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) if the device is not affected,
        * [`CheckStatus.FAIL`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) otherwise.

```